frame/contracts/proc-macro/src/lib.rs

```diff
@@ -181,14 +181,20 @@ impl HostFn {
 		};
 		let msg = "only #[version(<u8>)] or #[unstable] attribute is allowed.";
 		let span = item.span();
-		let item = match item {
+		let mut item = match item {
 			syn::Item::Fn(i_fn) => Ok(i_fn),
 			_ => Err(err(span, msg)),
 		}?;
 
-		let name = item.sig.ident.to_string();
-		let attrs: Vec<&syn::Attribute> =
-			item.attrs.iter().filter(|m| !m.path.is_ident("doc")).collect();
+		let attrs = &mut item.attrs;
+
+		let name = if attrs.iter().any(|a| a.path.is_ident("prefixed_alias")) {
+			"seal_".to_string() + &item.sig.ident.to_string()
+		} else {
+			item.sig.ident.to_string()
+		};
```

Comment:
You may be able to remove all this entirely, and do it all in `EnvDef::try_from`. You could set the `item.sig.ident` to the prefix, and just remove that attribute and pass straight through.